### DISCUSSION TITLE
homebrew has deprecated qt5

### DISCUSSION THREAD

**Discussion by sunderme** [texstudio-org/texstudio#1517] [General]
The consequence is that there is no poppler-qt5 build on homebrew and txs build on OSX fails.
Anybody interested to fix our CD build script to build poppler-qt5 as well ?

**Comment by muzimuzhi** [comment]
texstudio could maintain a homebrew tap itself which provides the required version (and dependences) of poppler.  As a test, I've tried extracting `poppler@21.03.0`, which uses qt5, to tap [`muzimuzhi/homebrew-versions`](https://github.com/muzimuzhi/homebrew-versions) and using it in texstudio, see commit https://github.com/muzimuzhi/texstudio/commit/ffcf49f1b47845003cfc2dc5e0003891018a8ce7. Waiting for the CD.

**Reply by muzimuzhi** [reply]
The "Configure" part was just passed, see https://github.com/muzimuzhi/texstudio/runs/2025284634. It is now building. So it seems the extracted `poppler@21.03.0` formula works.

**Comment by sunderme** [comment]
do you provide a PR (including the formula) ?

**Reply by muzimuzhi** [reply]
Let me see if the formula could be directly added to texstudio's main repo, instead of importing from a specific homebrew tap repo.

**Reply by muzimuzhi** [reply]
BTW, does the file `./travis-ci/mac/poppler.rb` still in use?

**Reply by sunderme** [reply]
No, it can be replaced

**Comment by muzimuzhi** [comment]
> do you provide a PR (including the formula) ?

See #1518

**Reply by sunderme** [reply]
Thanks for looking into it. I have pulled the PR and txs seems to build fine again.

**Comment by muzimuzhi** [comment]
Using `brew install poppler` as done in PR #1518, is as fragile as before. So when homebrew updates poppler to use qt6, then texstudio's CD will fail just like this time. If homebrew will provide two variants of poppler, that's great. If not, then texstudio will need to either build poppler with qt5 and cmake options `-DENABLE_QT5=ON -DENABLE_QT6=OFF`, or maintain a modified `poppler.rb`, and perhaps backups the build results for time saving.

For the second approach, I've tried in https://github.com/muzimuzhi/texstudio/commit/0c2e47ac4ff590bcfdb2e6533aaad65a0bddb2e5. Maybe this could be used when "that day" comes.
